lib/restforce/middleware/custom_headers.rb

```diff
@@ -0,0 +1,12 @@
+module Restforce
+  # Middleware that allows you to specify custom request headers
+  # when initializing Restforce client
+  class Middleware::CustomHeaders < Restforce::Middleware
+    def call(env)
+      headers = @options[:request_headers]
+      env[:request_headers].merge!(headers) if headers.is_a?(Hash)
```

Comment:
As it stands right now, we are allowing the user to overwrite whatever existing headers are present in the request (`env[:request_headers]`) with anything they pass in as `@option[:request_headers]`.  Are you saying we should flip it to 
```
headers.merge!(env[:request_headers])
```
so the user can't overwrite headers set elsewhere in the gem?

I'm totally fine with this, honestly I could argue in favor of going either direction, just making sure I understand what you're suggesting